src/Nethermind/Nethermind.Core/Transaction.cs

```diff
@@ -201,4 +202,21 @@ public interface ITransactionSizeCalculator
     {
         int GetLength(Transaction tx);
     }
+
+    /// <summary>
+    /// Holds network form fields for <see cref="TxType.Blob" /> transactions
+    /// </summary>
+    public class ShardBlobNetworkWrapper
+    {
+        public ShardBlobNetworkWrapper(byte[][] blobs, byte[][] commitments, byte[][] proofs)
+        {
+            Blobs = blobs;
+            Commitments = commitments;
+            Proofs = proofs;
+        }
+
+        public byte[][] Commitments { get; set; }
+        public byte[][] Blobs { get; set; }
+        public byte[][] Proofs { get; set; }
+    }
```

Comment:
I don't like it. We are potentially splitting all those arrays and not really do anything with them later. Actually we even flatten them later! This looks like a lot of unnecessary allocations and array manipulation.

Maybe we can keep it as either one or 3 arrays, and potentially come additional counts or offsets.

Reply:
Flattened arrays looked better with SSZ when you can just read all blobs in one chunk. I changed it with RLP, because RLP decoding becomes looking too polluted with DecodeAndFlattenByteArrays. Let's discuss it in terms of priorities today